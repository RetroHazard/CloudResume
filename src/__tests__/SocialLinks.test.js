import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import SocialLinks from '../components/social_links';

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => ({ children }) => {
    const generateRandomSocials = () => {
        const numberOfSocials = Math.floor(Math.random() * 5) + 1; // Random number between 1 and 5
        return {
            Socials: Array.from({ length: numberOfSocials }, (_, index) => ({
                display: true,
                name: `Social${index + 1}`,
                link: `http://social${index + 1}.com`,
                logo: 'mdi:home', // Mock icon identifier
            })),
        };
    };

    const data = generateRandomSocials();
    return children(data);
});

describe('SocialLinks Component', () => {
    test('renders the correct number of social links', () => {
        const { container } = render(<SocialLinks />);

        const socialLinks = screen.getAllByRole('link');
        const numberOfLinks = socialLinks.length;

        // Check that the number of links is between 1 and 5
        expect(numberOfLinks).toBeGreaterThanOrEqual(1);
        expect(numberOfLinks).toBeLessThanOrEqual(5);

        socialLinks.forEach((link, index) => {
            expect(link).toHaveAttribute('href', `http://social${index + 1}.com`);
            expect(link).toHaveAttribute('aria-label', `Social${index + 1}`);
            const icon = link.querySelector('.social-link');
            expect(icon).toBeInTheDocument();
            expect(icon).toHaveAttribute('icon', 'mdi:home');
        });
    });

    test('matches the snapshot', () => {
        const { asFragment } = render(<SocialLinks />);
        expect(asFragment()).toMatchSnapshot();
    });
});