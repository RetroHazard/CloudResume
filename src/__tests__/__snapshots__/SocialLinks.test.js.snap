// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex gap-3 flex-wrap sm:flex-nowrap"
  >
    <a
      aria-label="Social1"
      class="social-link"
      href="http://social1.com"
    >
      <i
        class="text-base text-content-icons"
      >
        <iconify-icon
          class="social-link"
          classname="social-link"
          height="1.25em"
          icon="mdi:home"
          width="1.25em"
        />
      </i>
    </a>
    <a
      aria-label="Social2"
      class="social-link"
      href="http://social2.com"
    >
      <i
        class="text-base text-content-icons"
      >
        <iconify-icon
          class="social-link"
          classname="social-link"
          height="1.25em"
          icon="mdi:home"
          width="1.25em"
        />
      </i>
    </a>
    <a
      aria-label="Social3"
      class="social-link"
      href="http://social3.com"
    >
      <i
        class="text-base text-content-icons"
      >
        <iconify-icon
          class="social-link"
          classname="social-link"
          height="1.25em"
          icon="mdi:home"
          width="1.25em"
        />
      </i>
    </a>
    <a
      aria-label="Social4"
      class="social-link"
      href="http://social4.com"
    >
      <i
        class="text-base text-content-icons"
      >
        <iconify-icon
          class="social-link"
          classname="social-link"
          height="1.25em"
          icon="mdi:home"
          width="1.25em"
        />
      </i>
    </a>
    <a
      aria-label="Social5"
      class="social-link"
      href="http://social5.com"
    >
      <i
        class="text-base text-content-icons"
      >
        <iconify-icon
          class="social-link"
          classname="social-link"
          height="1.25em"
          icon="mdi:home"
          width="1.25em"
        />
      </i>
    </a>
  </div>
</DocumentFragment>
`;
